pkg/execution/executor/executor.go

```diff
@@ -652,6 +690,29 @@ func (e *executor) Execute(ctx context.Context, id state.Identifier, item queue.
 
 		// Only just starting:  run lifecycles on first attempt.
 		if item.Attempt == 0 {
+			// NOTE:
+			// annotate the step as the first step of the function run.
+			// this way the delay associated with this run is directly correlated to the delay of the
+			// function run itself.
+			span.SetAttributes(attribute.Bool(consts.OtelSysStepFirst, true))
+
+			// Set the start time and spanID in metadata for subsequent runs
+			// This should be an one time operation and is never updated after,
+			// which is enforced on the Lua script.
+			update := state.MetadataUpdate{
+				Context:                   md.Context,
+				DisableImmediateExecution: md.DisableImmediateExecution,
+				SpanID:                    fnSpanID.String(),
+				StartedAt:                 &fnStartTime,
+			}
+			if item.RunInfo != nil {
+				// store in metadata as nano seconds for easier conversion later
+				update.SystemDelay = item.RunInfo.Latency.Nanoseconds()
```

Comment:
Is system delay what we're inferring anyway? I'm wondering if we need to include it here.

Reply:
We're not inferring this.